service/lib/dinstaller/status.rb

```diff
@@ -19,20 +19,47 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
-# YaST specific code lives under this namespace
 module DInstaller
-  # This class represents the installer status
-  class InstallerStatus
+  # Installer status
+  class Status
+    ERROR = 0.freeze
+    NOT_PROBED = 1.freeze
+    PROBING = 2.freeze
+    PROBED = 3.freeze
+    INSTALLING = 4.freeze
+    INSTALLED = 5.freeze
+
+    # @return [Integer]
     attr_reader :id
 
+    # Constructor
+    #
+    # @param id [Integer] initial status
     def initialize(id)
       @id = id
+      @on_change_callbacks = []
+    end
+
+    # Changes the status
+    #
+    # Callbacks are called.
+    #
+    # @param id [Integer] new status
+    def change(id)
+      @id = id
+      on_change_callbacks.each(&:call)
     end
 
-    ERROR = new(0)
```

Comment:
I like this old approach more (in which `InstallerStatus::ERROR` returns an status object) than the correct one (in which `Status::ERROR` returns a number).